common/src/Sync.php

```diff
@@ -0,0 +1,129 @@
+        $product = new Product($request_data['product_review']['product_id'], false, $lang_id);
+        if (!Validate::isLoadedObject($product)) {
+            throw new HttpException(404, sprintf('Could not find product with ID (%d)', $request_data['product_review']['product_id']));
+        }
+
+        if (!Configuration::get($this->module->getConfigName('SYNC_PROD_REVIEWS'))) {
+            throw new HttpException(403, 'Product review sync is disabled.');
+        }
+
+        $this->syncProductReview($request_data['product_review']);
+    }
+
+    /**
+     * @throws Exception
+     */
+    private function syncProductReview(array $product_review) {
+        $this->ajax = 1;
+        /** @var EntityManagerInterface $entityManager */
+        $entity_manager = $this->container->get('doctrine.orm.entity_manager');
+        $product_comment_entity = new ProductComment();
+        $date_add = DateTime::createFromFormat('Y-m-d H:i:s', $product_review['created']);
+
+        if ($product_review['id']) {
+            $product_comment_repository = $entity_manager->getRepository(ProductComment::class);
+            $product_comment = $product_comment_repository->find($product_review['id']);
+            if ($product_review['deleted']) {
+                $product_comment->setDeleted(1);
+                $this->logReviewSync($product_review['id'], true);
+            } else {
+                $product_comment->setDeleted(0);
+                $this->logReviewSync($product_review['id']);
+            }
+            $entity_manager->flush();
+            $this->ajaxRender(json_encode(['review_id' => $product_review['id']], JSON_PARTIAL_OUTPUT_ON_ERROR));
+            die();
+        }
+
+        $product_comment_entity->setProductId($product_review['product_id'])
+            ->setCustomerId($this->getCustomerIdByEmail($product_review['reviewer']['email']) ?? 0)
+            ->setGuestId(0)
+            ->setTitle($product_review['title'])
+            ->setContent($product_review['review'])
+            ->setCustomerName($product_review['reviewer']['name'])
+            ->setGrade($product_review['rating'])
+            ->setValidate(1)
+            ->setDateAdd($date_add);
+        $entity_manager->persist($product_comment_entity);
+        $entity_manager->flush();
+
+        $review_id = $product_comment_entity->getId();
+        $this->logReviewSync($review_id);
+        $this->ajaxRender(json_encode(['review_id' => $review_id], JSON_PARTIAL_OUTPUT_ON_ERROR));
+    }
+
+    private function getCustomerIdByEmail(string $email) {
+        $customer = new Customer();
+        $customer->getByEmail($email);
+
+        return $customer->id;
+    }
+
+    private function logReviewSync(string $review_id, bool $deleted = false) {
+        $deleted
+            ? \PrestaShopLogger::addLog(sprintf('Deleted product review with ID (%d)', $review_id))
+            : \PrestaShopLogger::addLog(sprintf('Saved product review with ID (%d)', $review_id));
```

Comment:
No need to repeat this twice if the only difference is Saved/Deleted

```suggestion
\PrestaShopLogger::addLog(sprintf('%s product review with ID (%d)', $deleted ? 'Deleted' : 'Saved', $review_id));
```